client/my-sites/domains/map-domain/test/map-domain.js

```diff
@@ -31,65 +43,82 @@ describe( 'MapDomain component', () => {
+		);
 		expect( pageSpy.redirect ).toBeCalledWith( '/domains/add/mapping' );
 	} );
 
 	test( 'renders a MapDomainStep', () => {
-		const wrapper = shallow( <MapDomain { ...defaultProps } /> );
-		expect( wrapper.find( MapDomainStep ) ).toHaveLength( 1 );
+		render( <MapDomain { ...defaultProps } /> );
+		expect( screen.queryByTestId( 'map-domain-step' ) ).toBeInTheDocument();
 	} );
 
-	test( "goes back when HeaderCake's onClick is fired", () => {
-		const wrapper = shallow( <MapDomain { ...defaultProps } /> );
-		expect( wrapper.find( HeaderCake ).prop( 'onClick' ) ).toEqual( wrapper.instance().goBack );
+	test( "goes back when HeaderCake's onClick is fired", async () => {
+		render( <MapDomain { ...defaultProps } /> );
+		// @TODO be more specific maybe to omit the `getALLByRole`
+		const [ backBtn ] = screen.getAllByRole( 'button', { name: /back/i } );
+		await userEvent.click( backBtn );
+		expect( pageSpy ).toHaveBeenCalledWith( `/domains/add/${ defaultProps.selectedSiteSlug }` );
 	} );
 
-	test( 'goes back to /domains/add if no selected site', () => {
-		const wrapper = shallow( <MapDomain { ...defaultProps } selectedSite={ null } /> );
-		wrapper.instance().goBack();
+	test( 'goes back to /domains/add if no selected site', async () => {
+		render( <MapDomain { ...defaultProps } selectedSite={ null } /> );
+		const [ backBtn ] = screen.getAllByRole( 'button', { name: /back/i } );
+		await userEvent.click( backBtn );
 		expect( pageSpy ).toBeCalledWith( '/domains/add' );
 	} );
 
-	test( 'goes back to domain management for VIP sites', () => {
-		const wrapper = shallow(
+	test( 'goes back to domain management for VIP sites', async () => {
+		render(
 			<MapDomain
 				{ ...defaultProps }
 				selectedSiteSlug="baba"
 				selectedSite={ { ...defaultProps.selectedSite, is_vip: true } }
 			/>
 		);
-		wrapper.instance().goBack();
+		const [ backBtn ] = screen.getAllByRole( 'button', { name: /back/i } );
+		await userEvent.click( backBtn );
 		expect( pageSpy ).toBeCalledWith( domainManagementList( 'baba' ) );
 	} );
 
-	test( 'goes back to domain add page if non-VIP site', () => {
-		const wrapper = shallow( <MapDomain { ...defaultProps } selectedSiteSlug="baba" /> );
-		wrapper.instance().goBack();
+	test( 'goes back to domain add page if non-VIP site', async () => {
+		render( <MapDomain { ...defaultProps } selectedSiteSlug="baba" /> );
+		const [ backBtn ] = screen.getAllByRole( 'button', { name: /back/i } );
+		await userEvent.click( backBtn );
 		expect( pageSpy ).toBeCalledWith( '/domains/add/baba' );
 	} );
 
 	test( 'does not render a notice by default', () => {
-		const wrapper = shallow( <MapDomain { ...defaultProps } /> );
-		// we match the notice by props, because enzyme isn't matching the Notice type for some reason
-		expect( wrapper.find( { status: 'is-error' } ) ).toHaveLength( 0 );
+		render( <MapDomain { ...defaultProps } /> );
+		expect( screen.queryByLabelText( 'Notice' ) ).toBeNull();
```

Comment:
Perhaps we can be a bit more specific about what we're asserting here?

```suggestion
		expect( screen.queryByLabelText( 'Notice' ) ).not.toBeInTheDocument();
```